src/osp/Active/SysRender.cpp

```diff
@@ -0,0 +1,275 @@
+{
+    Package& resources = rScene.get_context_resources();
+
+    using namespace osp::shader;
+    using namespace adera::shader;
+    using namespace Magnum;
+
+    /* Load shaders (eventually should happen elsewhere) */
+    resources.add<Phong>("phong_shader",
+        Phong{Magnum::Shaders::Phong::Flag::DiffuseTexture});
+
+    resources.add<PlumeShader>("plume_shader");
+
+    using Magnum::Shaders::MeshVisualizer3D;
+    resources.add<MeshVisualizer3D>("mesh_vis_shader",
+        MeshVisualizer3D{MeshVisualizer3D::Flag::Wireframe
+        | MeshVisualizer3D::Flag::NormalDirection});
+
+    resources.add<FullscreenTriShader>("fullscreen_tri_shader");
+
+    /* Generate fullscreen tri for texture rendering */
+    if (resources.get<GL::Mesh>("fullscreen_tri").empty())
+    {
+        Vector2 screenSize = Vector2{GL::defaultFramebuffer.viewport().size()};
+
+        float aspectRatio = screenSize.x() / screenSize.y();
+
+        std::array<float, 12> surfData
+        {
+            // Vert position    // UV coordinate
+            -1.0f,  1.0f,       0.0f,  1.0f,
+            -1.0f, -3.0f,       0.0f, -1.0f,
+             3.0f,  1.0f,       2.0f,  1.0f
+        };
+
+        GL::Buffer surface(std::move(surfData), GL::BufferUsage::StaticDraw);
+        GL::Mesh surfaceMesh;
+        surfaceMesh
+            .setPrimitive(Magnum::MeshPrimitive::Triangles)
+            .setCount(3)
+            .addVertexBuffer(std::move(surface), 0,
+                FullscreenTriShader::Position{},
+                FullscreenTriShader::TextureCoordinates{});
+        resources.add<GL::Mesh>("fullscreen_tri", std::move(surfaceMesh));
+    }
+
+    /* Configure pipeline resources */
+
+    // Add an offscreen framebuffer
+
+    Vector2i viewSize = GL::defaultFramebuffer.viewport().size();
+
+    GL::Texture2D color;
+    color.setStorage(1, GL::TextureFormat::RGB8, viewSize);
+    DependRes<GL::Texture2D> colorRes =
+        resources.add<GL::Texture2D>("offscreen_fbo_color", std::move(color));
+
+    GL::Renderbuffer depthStencil;
+    depthStencil.setStorage(GL::RenderbufferFormat::Depth24Stencil8, viewSize);
+    DependRes<GL::Renderbuffer> depthStencilRes =
+        resources.add<GL::Renderbuffer>("offscreen_fbo_depthStencil", std::move(depthStencil));
+
+    GL::Framebuffer fbo(Range2Di{{0, 0}, viewSize});
+    fbo.attachTexture(GL::Framebuffer::ColorAttachment{0}, *colorRes, 0);
+    fbo.attachRenderbuffer(GL::Framebuffer::BufferAttachment::DepthStencil, *depthStencilRes);
+
+    DependRes<GL::Framebuffer> fboRes =
+        resources.add<GL::Framebuffer>("offscreen_fbo", std::move(fbo));
+
+    rScene.reg_emplace<ACompRenderTarget>(rScene.hier_get_root(),
+        viewSize, std::move(fboRes));
+    rScene.reg_emplace<ACompFBOColorAttachment>(rScene.hier_get_root(), colorRes);
+
+
+    // Create the default render pipeline
+    resources.add<RenderPipeline>("default",
+        std::move(create_render_pipeline()));
```

Comment:
if create_render_pipeline returns an object by value, or by const&, then this std::move() will have no effect.

Functions that return an object by value will automatically have the returned value converted to && in this context.

std::move() silently ignores const& parameters, returning a const&.

So the only time this std::move() will have an effect on the program is if create_render_pipeline() returns a value by non-const reference, and while that's not inherently problematic, given the name of the function, i'd argue that's not what this function should return.

So, long story short, the std::move() here can be removed.

Reply:
Good to know, thanks